tests/library/capabilities.spec.ts

```diff
@@ -270,3 +276,41 @@ it('requestFullscreen', async ({ page, server, browserName, headless, isLinux })
   });
   expect(await page.evaluate(() => !!document.fullscreenElement)).toBeFalsy();
 });
+
+it('should send no Content-Length header for GET requests with a Content-Type', async ({ page, server, browserName }) => {
+  it.info().annotations.push({ type: 'issue', description: 'https://github.com/microsoft/playwright/issues/22569' });
+  it.skip(browserName === 'webkit' && hostPlatform === 'ubuntu20.04', 'libsoup2.4 bug');
+  it.fixme(browserName === 'webkit' && hostPlatform === 'ubuntu22.04', 'waiting for libsoup3 upgrade');
+  await page.goto(server.EMPTY_PAGE);
+  const [request] = await Promise.all([
+    server.waitForRequest('/empty.html'),
+    page.evaluate(() => fetch('/empty.html', {
+      'headers': { 'Content-Type': 'application/json' },
+      'method': 'GET'
+    }))
+  ]);
+  expect(request.headers['content-length']).toBe(undefined);
+});
+
+it('should infer a double encoded gzip response content-type from the script tag', async ({ page, server, browserName }) => {
```

Comment:
Not sure we want it as the behavior varies between browsers and platforms and we won't be fixing it if it breaks.